【UnitTestFix No.1】fix test_activation_op.py

scyyh11 opened the pull request:
<!-- TemplateReference: https://github.com/PaddlePaddle/Paddle/wiki/PULL-REQUEST-TEMPLATE--REFERENCE -->
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->

### PR Category
<!-- One of [ User Experience | Execute Infrastructure | Operator Mechanism | CINN | Custom Device | Performance Optimization | Distributed Strategy | Parameter Server | Communication Library | Auto Parallel | Inference | Environment Adaptation ] -->

Operator Mechanism

### PR Types
<!-- One of [ New features | Bug fixes | Improvements | Performance | BC Breaking | Deprecations | Docs | Devs | Not User Facing | Security | Others ] -->

Bug fixes

### Description
<!-- Describe what you’ve done -->

修复了test_activation_op.py中不可用的test。
删除了SoftRelu相关的测试，因为2.6版本后已经不支持SoftRelu。

<img width="778" height="209" alt="ed2cb0a17a8180c8fbec01c0c2607457" src="https://github.com/user-attachments/assets/ed7a0ee3-59d2-4b77-9b91-9652628664c6" />

@luotao1 @YqGe585

scyyh11 committed:
fix: eliminate warning "API "paddle.base.dygraph.tensor_patch_methods.gradient" is deprecated since 2.1.0, and will be removed in future versions. Reason: Please use tensor.grad, which returns the tensor value of the gradient."

scyyh11 committed:
fix: skip unsupported integer gradient checks for ceil/floor prim tests

scyyh11 committed:
fix: Added paddle.pir_utils import and wrapped legacy-only tests (TestPow_API, TestRelu6APIWarnings) with OldIrGuard plus fresh Program guards. Adjusted TestRelu_NanInput to convert the NaN-count tensor to a host scalar before asserting, sidestepping PIR’s static bool(Tensor) restriction.

scyyh11 committed:
fix: improve activation tests for PIR compatibility and shape handling

- Fix shape comparison in TestSinhAPI and TestCoshAPI by converting shapes to lists
- Disable gradient check for TestRelu_NanInput class to handle NaN input cases
- Refactor TestSqrtOutAndAlias to use PIR-compatible API with positional arguments
- Simplify test execution by removing unnecessary startup program call
- Update variable naming and data feeding for better PIR support

scyyh11 committed:
fix: improve activation op tests for type compatibility and PIR support

- Enable int32 input support for sqrt, tanh, sinh, cosh ops with auto-cast to float32
- Fix shape comparison in TestTanAPI by converting shapes to lists
- Refactor TestRelu_NanInput to support both static and dygraph execution modes
- Update test comments to reflect new int32 input support capabilities

scyyh11 committed:
refactor: remove TestSoftRelu class from activation tests

- Deleted the TestSoftRelu class to streamline activation operation tests.
- Updated test creation calls to exclude TestSoftRelu for both FP16 and BF16 classes.

scyyh11 committed:
fix: update TestRelu_NanInput to prevent base class method call

- Added a test_check_output method to override the base class behavior.
- Refactored NaN count calculation to use numpy's isnan method for clarity.

scyyh11 committed:
fix: update activation op tests to disable check_prim_pir

- Set check_prim_pir to False in TestSigmoidBF16 and TestPow classes to improve compatibility with PIR.
- Adjusted test configurations to ensure consistent behavior across activation operation tests.

scyyh11 committed:
Merge branch 'PaddlePaddle:develop' into fix/test_activation_op

scyyh11 committed:
Merge branch 'PaddlePaddle:develop' into fix/test_activation_op

scyyh11 committed:
fix: correct TestPow FP16 prim checker configuration

The TestPow FP16 test was failing because it incorrectly expected the pow
operation to be decomposed in PIR mode (check_prim_pir=True). However,
pow is a primitive operation and should not be decomposed. Changed the
configuration to check_prim_pir=False to match the primitive nature
of the pow operation.

scyyh11 committed:
Merge branch 'PaddlePaddle:develop' into fix/test_activation_op

scyyh11 committed:
fix:
- Remove unnecessary comments and clean up code.
- Adjusted assertions in TestPow_API for clarity and consistency.

scyyh11 committed:
refactor: optimize import

YqGe585 commented on test/legacy_test/test_activation_op.py:
```diff
@@ -3437,6 +3459,10 @@ def test_warnings(self):
         with (
             static_guard(),
             warnings.catch_warnings(record=True) as context,
+            paddle.pir_utils.OldIrGuard(),
```
OldIr相关case可以直接移除，Paddle不再维护OldIr相关机制。

SigureMo commented on test/legacy_test/test_activation_op.py:
```diff
@@ -534,7 +534,7 @@ def test_check_output(self):
             check_prim=False,
             check_pir=True,
-            check_prim_pir=True,
+            check_prim_pir=False,
```
`check_prim` 可以关，`check_prim_pir` 不能关呀，PIR 下有问题应该去修复

scyyh11 replied:
您好，我在修复这个部分的时候遇到了如下的问题：

* 在 **`primitive.yaml` 和 `primitive_ops.h`** 里，`sigmoid` 和 `pow` 被定义为 **primitive 算子**。
* 但在 **`composite_rules.py`** 里，它们又有明确的 **分解规则**（比如 sigmoid → `1/(1+exp(-x))`）。
* 在测试时，虽然我调用了 decomposition pass，但这两个算子并不会真的被分解，操作列表保持不变，导致断言失败。

所以我现在的问题是，对于pow和sigmoid，是要把他们作为primitive 算子来对待么

SigureMo replied:
组合算子的细节需要节后 @xiaoguoguo626807 确认下

xiaoguoguo626807 replied:
这个理论上这两个算子是组合算子，但是因为拆解之后会引起精度或者性能问题，所以实际使用的时候不会拆解

xiaoguoguo626807 replied:
你描述的问题是旧ir 下遇到的吗，提到的文件都是旧ir的

scyyh11 replied:
> 你描述的问题是旧ir 下遇到的吗，提到的文件都是旧ir的

我是在做旧IR到新IR的迁移，修复部分因为框架更新不能用的test

scyyh11 committed:
fix: remove OldIr related test case. revert modified check_prim_pir back to True for sigmoid and pow test(need fix).

scyyh11 committed:
fix:
- skip check_static_comp for prim operator in prim_op_test.py
- refactor the prim_op_type for TestPow and TestSigmoidBF16 in test_activation_op.py since they are both primitive operators.

xiaoguoguo626807 commented on test/legacy_test/prim_op_test.py:
```diff
@@ -1080,11 +1080,11 @@ def check_eager_comp(self):
             core.set_prim_eager_enabled(False)
 
     def check_static_comp(self):
+        # prim op don't need gradient decomposition
```
这里不对吧，prim_op 是需要测试反向拆解的

scyyh11 replied:
好的

xiaoguoguo626807 commented on test/legacy_test/test_activation_op.py:
```diff
@@ -504,7 +501,7 @@ def init_shape(self):
 class TestSigmoidBF16(OpTest):
     def setUp(self):
         self.op_type = "sigmoid"
-        self.prim_op_type = "comp"
```
这个算子比较特殊，可以测comp , 测试前向拆解正确性， 也可以测prim, 测反向拆解正确性，严格来说都需要测一下

scyyh11 committed:
fix: reset check_static_comp

scyyh11 committed:
fix: add new TestSigmoidFp32_Comp to verify forward decomposition correctness
  of sigmoid under FP32

scyyh11 committed:
fix: add TestPowFp64_Comp to verify forward decomposition correctness and gradient checks for pow operation in FP64 precision

SigureMo commented on test/legacy_test/test_activation_op.py:
```diff
@@ -5987,7 +6016,6 @@ def test_check_grad(self):
 )
 create_test_act_fp16_class(TestBRelu, check_pir=True)
 create_test_act_fp16_class(TestRelu6)
-create_test_act_fp16_class(TestSoftRelu, check_dygraph=False)
```
`TestSoftRelu` 删掉的理由是什么？https://github.com/PaddlePaddle/Paddle/pull/75553/commits/309031363f65da2e4cef58c03269253c5d9d92d1 提到 `Deleted the TestSoftRelu class to streamline activation operation tests.` 是说 `TestSoftRelu` 和别的有重复么？相关改动原因最好在 PR 描述里说明下

scyyh11 replied:
您好，删除SoftRelu相关测试的原因是在2.6版本之后SoftRelu在官方文档里已经搜不到相关内容，保留会遇到SoftRelu算子未注册的报错，我和@YqGe585 沟通确认后删除了相关的测试。

xiaoguoguo626807 reviewed (approved):
LGTM for decomp test

YqGe585 reviewed (approved):
LGTM
